### ISSUE TITLE
Exclude all attributes starting with "goog_" from outbound Pub/Sub headers 

### ISSUE THREAD

**Issue by elefeint** [GoogleCloudPlatform/spring-cloud-gcp#776]
Report came from Gitter -- in a Spring Cloud Stream application, `HeaderMapper` fields are not customizable in the adapters. In a Processor SCS application, if the Pub/Sub message received from one subscription is passed as-is to a topic, the message will be rejected with:
```
com.google.api.gax.rpc.InvalidArgumentException: io.grpc.StatusRuntimeException: INVALID_ARGUMENT: The request contains an attribute key that is not valid (key=googclient_deliveryattempt). Attribute keys must be non-empty and must not begin with 'goog' (case-insensitive).
```

To prevent this issue, `PubSubHeaderMapper.outboundHeaderPatterns` needs to include `"!goog_*` by default.

**Comment by ddixit14** [comment]
Sounds like something I can take up next? @elefeint

**Comment by elefeint** [comment]
Yes, that's a good idea. First, reproduce the issue by creating a sample application that has a Spring Cloud Stream *processor* -- our current sample only has a source and a sink.

**Comment by elefeint** [comment]
@ddixit14 Hints on reproducing using `spring-cloud-gcp-pubsub-stream-functional-sample`:
- `googclient_deliveryattempt` only exists when the subscription has DLQ enabled. Can't reproduce otherwise.
- In Spring Cloud Stream, to turn something from a Sink to a Processor, the functional bean needs to switch from being a `Consumer` (one input, no outputs) to being a `Function` (one input, one output).
- Once you have a processor/function, tie the outgoing stream to a physical Pub/Sub topic with this property: `spring.cloud.stream.bindings.logUserMessage-out-0.destination=my-topic2`

Bonus question to think about: do we want to strip incoming `goog*` headers, outgoing ones, or both?
